cellengine/resources/fcs_file.py

```diff
@@ -223,12 +228,11 @@ def events(self, events):
 
     @property
     def spill_string(self):
-        if self._spill_string:
-            return self._spill_string
-        else:
-            ss = ce.APIClient().get_fcs_file(self.experiment_id, self._id).spill_string
-            self._spill_string = ss
-            return ss
+        if not self._spill_string and self.has_file_internal_comp:
+            self._spill_string = ce.APIClient().get_fcs_file(
+                self.experiment_id, self._id, as_dict=True
+            )["spillString"]
+        return self._spill_string or ""
```

Comment:
I think you can/should get rid of the `or ""` now.

Reply:
I kept it to silence the typing error in line 205, as `Compensation.from_spill_string()` wants a `str`, not `None`. It seems wrong to change `.from_spill_string` to accept `None`. Thoughts?

Reply:
I think I'd override mypy then... I don't see a way to declare class members types that depend on each other (i.e. `has_file_internal_comp` means `spill_string` is Str), do you know a way? Otherwise I guess just `ignore` it.

Reply:
This is still pending, right? I'm suggesting removing `or ""` and adding `# type: ignore` to line 253R.

Reply:
:+1: